src/Parse/ParsePolygon.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Parse;
+
+use Parse\Internal\Encodable;
+use Parse\ParseGeoPoint;
+
+/**
+ * ParsePolygon - Representation of a Parse Polygon object.
+ *
+ * @author Diamond Lewis <findlewis@gmail.com>
+ */
+class ParsePolygon implements Encodable
+{
+    /**
+     * The coordinates.
+     *
+     * @var array
+     */
+    private $coordinates;
+
+    /**
+     * Create a Parse Polygon object.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     */
+    public function __construct($coords)
+    {
+        $this->setCoordinates($coords);
+    }
+
+    /**
+     * Set the Coordinates value for this Polygon.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     *
+     * @throws ParseException
+     */
+    public function setCoordinates($coords)
+    {
+        if (count($coords) < 3 || !is_array($coords)) {
```

Comment:
Can you move the array exception here into a separate check just above? It's close but the user should be informed that an array of coordinates is required when one is missing, just to be more specific about which is which.